src/utils/getVisiblePeople.tsx

```diff
@@ -0,0 +1,76 @@
+import { Person } from '../types/Person';
+import { Century } from '../types/Filters';
+import { TableHeader } from '../types/TableHeader';
+
+type VisiblePeople = {
+  people: Person[];
+  selectedGender: string | null;
+  query: string;
+  selectedCenturies: Century[] | string[];
+  sortBy: string | null;
+  isReversed: string | null;
+};
+
+export const getVisiblePeople = ({
+  people, selectedGender, query, selectedCenturies, sortBy, isReversed,
+}: VisiblePeople) => {
+  let visiblePeople = [...people];
+
+  if (selectedGender) {
+    visiblePeople = visiblePeople.filter(({ sex }) => {
+      switch (selectedGender) {
+        case 'm':
+          return sex === selectedGender;
+
+        case 'f':
+          return sex === selectedGender;
+
+        default:
+          return sex;
+      }
+    });
+  }
+
+  if (query) {
+    const normalizedQuery = query.toLowerCase();
+
+    visiblePeople = visiblePeople.filter(person => (
+      person.name.toLowerCase().includes(normalizedQuery)
```

Comment:
looks like we can create separated variable for `toLowerCase().includes(normalizedQuery)` and reuse